### ISSUE TITLE
Add lexer/syntax highlighting to visually distinguish the different components of a command

### ISSUE THREAD

**Issue by donnemartin** [awslabs/aws-shell#27]
A lexer would make it easier to visually distinguish the different components of a command:

     aws [options] <command> <subcommand> [parameters]

## Config File

Default theme: vim

```
# visual theme. possible values: manni, igor, xcode, vim,
# autumn,vs, rrt, native, perldoc, borland, tango, emacs,
# friendly, monokai, paraiso-dark, colorful, murphy, bw,
# pastie, paraiso-light, trac, default, fruity.
# to disable themes, set theme = none
theme = vim

```

## Disable Syntax Highlighting

     theme = none

## Sample: Dark Background

     theme = vim
![Imgur](http://i.imgur.com/SUcBeDr.png)

## Samples: Light Background

     theme = vim

*Note: My terminal colors are customized which might be why we see light gray for `aws>`.*

![Imgur](http://i.imgur.com/8ZiQ332.png)

     theme = xcode

![Imgur](http://i.imgur.com/LpzLDBL.png)

**Comment by donnemartin** [comment]
I updated the initial comment with the proposed UI.

**Comment by donnemartin** [comment]
Implementation merged in 9149483166ac1c294aa3e8a2392c3d350e7ee852.
